pyodide-build/pyodide_build/mkpkg.py

```diff
@@ -119,11 +166,106 @@ def run_prettier(meta_path):
     subprocess.run(["npx", "prettier", "-w", meta_path])
 
 
+def _make_project_compare_name(project_name):
+    project_name_cleaned = project_name.replace("_", "").replace("-", "").lower()
+    if project_name_cleaned.startswith("py"):
+        project_name_cleaned = project_name[2:]
+    return project_name_cleaned
+
+
+# download a package from pypi and guess what modules are provided by it
+def _find_imports_from_package(url, project_name):
```

Comment:
Couldn't you just read `top_level.txt` from the wheel?

Reply:
I imagine the issue is that it wouldn't work with .tar.gz / .zip files.

Though we could use `top_level.txt` if it exists and otherwise fallback to  [`setuptools.find_packages`](https://setuptools.pypa.io/en/latest/userguide/package_discovery.html#finding-simple-packages). It's already a build dependency. This would require unzipping files into some temporary folder so it would be slower, but a least we wouldn't have to maintain that code.

Overall it could be good to split this function into,
 - a function that downloads the package to some path
 - a function that finds imports from either an archive or a folder (if we go with `find_packages`).